recipe-server/normandy/base/genome.py

```diff
@@ -0,0 +1,138 @@
+import json
+import math
+import random
+import string
+
+from hashlib import sha256
+
+
+class Genome:
+    """A seedable source that provides arbitrarily sized chunks of randomness."""
+    def __init__(self, chromosome, entropy):
+        self.chromosome = chromosome
+        self.entropy = entropy
+        self.log = []
+
+    @staticmethod
+    def generate(seed=random.random()):
+        """Asynchronously returns a Genome based on the given seed.
+
+        seed -- Any JSON serializable value. Will be used to seed the
+                genome. If not provided, will be randomly generated.
+        """
+        hash = sha256(json.dumps(seed).encode('utf-8'))
```

Comment:
I think the Python version of this should just call `str()` on seed, instead of using JSON, since Python is better at converting things to strings by default.